tensorflow/contrib/learn/python/learn/tests/test_saver.py

```diff
@@ -83,6 +83,20 @@ def testNoCheckpoints(self):
     with self.assertRaises(ValueError):
       learn.TensorFlowEstimator.restore(path)
 
+  def testOverwriteParams(self):
+    path = tf.test.get_temp_dir() + '/tmp_saver3'
+    random.seed(42)
+    iris = datasets.load_iris()
+    classifier = learn.TensorFlowDNNClassifier(hidden_units=[10, 20, 10],
+                                               n_classes=3,
+                                               learning_rate=0.1)
+    classifier.fit(iris.data, iris.target)
+    classifier.save(path)
+    self.assertEqual(classifier.learning_rate, 0.1)
+    new_classifier = learn.TensorFlowEstimator.restore(path,
+      new_params={'learning_rate': 0.2})
+    self.assertEqual(type(new_classifier), type(classifier))
+    self.assertEqual(new_classifier.learning_rate, 0.2)
```

Comment:
this not going to work the way you expect. for example learning rate is now captured in the variable - so resetting it on the class - won't help at this point to get a different learning rate.

We are moving an Estimator class that uses graph_actions, which re-creates graph each time `fit`/`predict` is called. That way one will be able to override parameters.